longevity_oos_test.py

```diff
@@ -0,0 +1,365 @@
+        return all(self.verify_stress_thread(stress) for stress in stress_queue)
+
+    def run_write_stress(self):
+        stress_queue = []
+        self.assemble_and_run_all_stress_cmd(stress_queue, self.params.get(
+            'stress_cmd_w'), self.params.get('keyspace_num'))
+        return all(self.verify_stress_thread(stress) for stress in stress_queue)
+
+    def scale_out(self, nr_nodes=None, rack=None):
+        """
+        Scale out the cluster by adding new nodes.
+
+        :param nr_nodes: Number of nodes to add, defaults to one per rack.
+        :param rack: The rack to add the nodes to, defaults to round robin through the available racks.
+        """
+        instance_type = self.params.get("instance_type_db")
+        nr_nodes = nr_nodes or self.db_cluster.racks_count
+        added_nodes = self.db_cluster.add_nodes(
+            count=nr_nodes, instance_type=instance_type, enable_auto_bootstrap=True, rack=rack)
+        self.monitors.reconfigure_scylla_monitoring()
+        up_timeout = MAX_TIME_WAIT_FOR_NEW_NODE_UP
+        with adaptive_timeout(Operations.NEW_NODE, node=self.db_cluster.data_nodes[0], timeout=up_timeout):
+            self.db_cluster.wait_for_init(node_list=added_nodes, timeout=up_timeout, check_node_health=False)
+        self.db_cluster.set_seeds()
+        self.db_cluster.update_seed_provider()
+        self.db_cluster.wait_for_nodes_up_and_normal(nodes=added_nodes)
+        for node in self.db_cluster.nodes:
+            wait_no_tablets_migration_running(node, timeout=7200)
+
+    def get_compactions(self, node: BaseNode, interval: int) -> int:
+        """
+        Get the number of compactions that occurred on the node during the specified interval.
+
+        :param node: The node to get the compactions for.
+        :param interval: The time interval for the query, as last X seconds.
+        :return: The number of compactions that occurred on the node during the specified interval.
+        """
+        compaction_query = f'sum(scylla_compaction_manager_compactions{{instance=~".*?{node.private_ip_address}.*?"}})'
+        now = time()
+        results = self.prometheus_db.query(query=compaction_query, start=now - interval, end=now)
+        self.log.info(f"Compactions on node {node.name} ({node.private_ip_address}): {results}")
+        # results is of this form:
+        # [{'metric': {}, 'values': [[1749638594.936, '0'], [1749638614.936, '0'], [1749638634.936, '0'], [1749638654.936, '0']]}]
+        return sum(int(v[1]) for v in results[0]['values']) if results else 0
+
+    def test_oos_write(self):
+        """
+        Fill the cluster to 90%
+        Start another write, that would need more space than available
+        Write should fail, but the cluster should not run out of space
+        Scale out the cluster
+        Continue writing data
+        Write should succeed after scale out
+        """
+        self.run_prepare_write_cmd()
```

Comment:
Shouldn't we check here the disk utilization? The docstring says that the cluster is filled to 90%. So we rely on the properly configured `prepare_write_cmd` in a yaml file. It's error prone. We could easily run the misconfigured test if you change the instance type and forget to adjust `c-s`, or if you miscalculate.